include/zephyr/senss/senss_datatypes.h

```diff
@@ -0,0 +1,313 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSS_DATATYPES_H_
+#define SENSS_DATATYPES_H_
+
+#include <stdint.h>
+
+/**
+ * @brief Data Types
+ * @addtogroup senss_datatypes
+ * @{
+ */
+
+/**
+ * @struct senss_sensor_value_header
+ * @brief sensor value header
+ *
+ * Each sensor value data structure should have this header
+ *
+ * Here use 'base_timestamp' (uint64_t) and 'timestampe_delta' (uint32_t) to
+ * save memory usage in batching mode.
+ *
+ * The 'base_timestamp' is for readings[0], the 'timestamp_delta' is relation
+ * to the previous 'readings'. So,
+ * timestamp of readings[0] is
+ *	header.base_timestamp + readings[0].timestamp_delta.
+ * timestamp of readings[1] is
+ *	timestamp of readings[0] + readings[1].timestamp_delta.
+ *
+ * Since timestamp unit is micro seconds, the max 'timestamp_delta' (uint32_t)
+ * is 4295 seconds.
+ *
+ * If a sensor has batched data where two consecutive readings differ by
+ * more than 4295 seconds, the sensor subsystem core will split them
+ * across multiple instances of the readings structure, and send multiple
+ * events.
+ *
+ * This concept is borrowed from CHRE:
+ * https://cs.android.com/android/platform/superproject/+/master:\
+ * system/chre/chre_api/include/chre_api/chre/sensor_types.h
+ */
+struct senss_sensor_value_header {
+	/** base timestamp of this data readings, unit is micro seconds */
+	uint64_t base_timestamp;
+	/** count of this data readings */
+	uint16_t reading_count;
+};
+
+/**
+ * @brief Sensor value data structure types based on common data types.
+ * Suitable for common sensors, such as IMU, Light sensors and orientation sensors.
+ *
+ * Sensor subsystem will use scaled fixed point data structure for all sensor values,
+ * aligned the HID spec, using the format v*10^x to present the decimal value,
+ * where the v is integer number, either int8/uint8, int16/uint6, or int32/uint32, depends on
+ * required sensor data precision.
+ *
+ * The scale unit exponent x is int8 type int8 type with encoding meanings (page 68 of
+ * HID spec https://usb.org/sites/default/files/hutrr39b_0.pdf):
+ *
+ * Unit Exponet argument       Power of Ten (Scientific Notation)
+ * 0x00			       1 * 10E0
+ * 0x01			       1 * 10E1
+ * 0x02			       1 * 10E2
+ * 0x03			       1 * 10E3
+ * 0x04			       1 * 10E4
+ * 0x05			       1 * 10E5
+ * 0x06			       1 * 10E6
+ * 0x07			       1 * 10E7
+ * 0x08			       1 * 10E-8
+ * 0x09			       1 * 10E-7
+ * 0x0A			       1 * 10E-6
+ * 0x0B			       1 * 10E-5
+ * 0x0C			       1 * 10E-4
+ * 0x0D			       1 * 10E-3
+ * 0x0E			       1 * 10E-2
+ * 0x0F			       1 * 10E-1
+ *
+ * So, we can have below data present ranges:
+ *
+ * int8 type v:    [-128, 127] *10^[-8, 7]
+ * uint8 type v:   [0,	255] * 10^[-8, 7]
+ * int16 type v:   [-32768, 32767] * 10^[-8, 7]
+ * uint16 type v:  [0,	65535\r] * 10^[-8, 7]
+ * int32 type v:   [-2147483648,  2147483647] * 10^[-8, 7]
+ * uint32 type v:  [0,	4294967295] * 10^[-8, 7]
```

Comment:
I'd propose including 64-bit values here as well, even if we make it an optional KConfig flag, simply because 32-bits may be too small for some representations of data where high accuracy is required.

Reply:
Yes, added int64/uint64 values, and can easily adding new value structure data types for new sensor types.